### PR TITLE
feat: Request Logger & Logging context

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Smaller version of #7238 . This PR doesn't include any of the request timing info like the previous PR.

Please see the previous PR on details about what this PR is doing!

## Screenshots

image

### CODE REVIEW

File: dotcom-rendering/src/server/lib/logging.ts [hunk 1]
```diff
@@ -22,6 +27,7 @@ const logFields = (logEvent: LoggingEvent): unknown => {
 		'@version': 1,
 		level: logEvent.level.levelStr,
 		level_value: logEvent.level.level,
+		request,
```
Comment (AshCorr) [inline dotcom-rendering/src/server/lib/logging.ts:30]: Nice to have request details such as Page ID on log entries even if they're not generated by the request logger.

File: dotcom-rendering/src/server/lib/logging-store.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		type?: string;
+		platform?: string;
+	};
+	error?: {
+		message?: string;
+		stack?: string;
+	};
+};
+
+export const loggingStore = new AsyncLocalStorage<DCRLoggingStore>();
```
Comment (OllysCoding) [inline dotcom-rendering/src/server/lib/logging-store.ts:28]: Does this somehow understand the call stack that allows it to know exactly which request it is handling even without prop drilling?! I'd love to understand what's going on under the hood in node.js to make this happen!

**Review by OllysCoding — APPROVED** [review]
+1 This is awesome, can't pretend I understand what's going on under the hood yet, but nonetheless a massive win for DCR logging :)))

File: dotcom-rendering/src/server/lib/logging-store.ts (hunk 2 above)
Comment (AshCorr) [reply dotcom-rendering/src/server/lib/logging-store.ts:28]: Yes! Assuming I'm understanding the docs correctly! It was a new class introduced in Node 16! Their example usecase is pretty much exactly what we're using it for.

https://nodejs.org/api/async_context.html#introduction

Afraid I'm not exactly sure how it works under the hood, nodejs dark magic!

**Pull request merged**